src/plugins/discover/public/application/main/hooks/use_discover_state.ts

```diff
@@ -60,19 +60,11 @@ export function useDiscoverState({
   const stateContainer = useMemo(
     () =>
       getState({
```

Comment:
this is the main change in this PR, instead of passing in a function (`getStateDefaults`) that makes also use of a same named function `getStateDefaults`, now just `savedSearch` is being used the generate the default state in the state container.